app/code/Magento/Security/Model/UserExpiration.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Security\Model;
+
+/**
+ * Admin User Expiration model.
+ * @method string getUserId()
+ * @method \Magento\Security\Model\UserExpiration setUserId($userId)
+ * @method string getExpiresAt()
+ * @method \Magento\Security\Model\UserExpiration setExpiresAt($value)
+ */
+class UserExpiration extends \Magento\Framework\Model\AbstractModel
```

Comment:
This model has no interface. The fields are defined in phpdoc. Suggest to introduce and use the interface for expiration model

Reply:
@lfolco

Reply:
Do you mean create an interface class or define the methods in `UserExpiration`?

Reply:
Just a regular php interface with implementation inside UserExpiration class.

E.g. 
```
interface UserExpirationInterface {
    public function getExpiresAt(): string, date or timestamp type
}

class UserExpiration implements UserExpirationInterface {
public function getExpiresAt() {
 return 'something';
}


//Some other place

class OtherClass {
    public function save(UserExpirationInterface $expiration) {
         $expiration->getExpiresAt();
    }

}
```

Reply:
How would I use that in `\Magento\Security\Observer\AfterAdminUserSave`, for example? This relies on the abstract model/resource model:

```
public function execute(Observer $observer)
    {
        /* @var $user \Magento\User\Model\User */
        $user = $observer->getEvent()->getObject();
        if ($user->getId()) {
            $expiresAt = $user->getExpiresAt();
            /** @var \Magento\Security\Model\UserExpiration $userExpiration */
            $userExpiration = $this->userExpirationFactory->create();
            $this->userExpirationResource->load($userExpiration, $user->getId());

            if (empty($expiresAt)) {
                // delete it if the admin user clears the field
                if ($userExpiration->getId()) {
                    $this->userExpirationResource->delete($userExpiration);
                }
            } else {
                if (!$userExpiration->getId()) {
                    $userExpiration->setId($user->getId());
                }
                $userExpiration->setExpiresAt($expiresAt);
                $this->userExpirationResource->save($userExpiration);
            }
        }
    }
```
Unless I create a repository as well? Or should I not use the new interface in the scenario where I need to load or save the UserExpiration object directly?

Reply:
So I guess I don't need to do this anymore? I was only asking for clarification, I would have preferred to implement it. I never heard back about my question.

Reply:
Hi @lfolco. Sorry for the delay and this accident, I start working on this PR and really want to delivery it. 
That is ok that you extend \Magento\Framework\Model\AbstractModel, all that new need it create the interface for this class. We should have a strict contract, currently, contract provided just in phpdoc block, it isn't strict and we have just one option to provide strict contract - it's an interface